File: 02/one.py
#!/bin/env python3

location = {'position': 0, 'depth': 0}
dirs = {
    'forward': ('position', int),
    'down': ('depth', int),
    'up': ('depth', lambda x: -int(x))
}

with open('input.txt') as infile:
    for line in infile:
        direction, distance = line.split()
        axis, parse = dirs[direction]
        location[axis] += parse(distance)

print(location['position'] * location['depth'])

File: 02/one.nim
import std/strutils

var
  position = 0
  depth = 0
  direction: string
  distance: string

for line in "input.txt".lines:
  (direction, distance) = line.split
  case direction
  of "forward":
    position.inc distance.parse_int
  of "down":
    depth.inc distance.parse_int
  of "up":
    depth.dec distance.parse_int

(position * depth).echo
